plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestTimestamp.java

```diff
@@ -79,38 +81,42 @@ public void testTimestampBackedByInt64()
                     Optional.of(parquetSchema),
                     false);
 
-            Iterator<SqlTimestamp> expectedValues = timestamps.build().iterator();
-            try (ConnectorPageSource pageSource = StandardFileFormats.TRINO_PARQUET.createFileFormatReader(session, HDFS_ENVIRONMENT, tempFile.getFile(), columnNames, ImmutableList.of(TIMESTAMP_MILLIS))) {
-                // skip a page to exercise the decoder's skip() logic
-                Page firstPage = pageSource.getNextPage();
-
-                assertTrue(firstPage.getPositionCount() > 0, "Expected first page to have at least 1 row");
+            testReadingAs(TIMESTAMP_MILLIS, session, tempFile, columnNames, timestampsMillis);
+        }
+    }
 
-                for (int i = 0; i < firstPage.getPositionCount(); i++) {
-                    expectedValues.next();
-                }
+    private <T> void testReadingAs(Type type, ConnectorSession session, ParquetTester.TempFile tempFile, List<String> columnNames, ImmutableList.Builder<T> expectedValues)
```

Comment:
`<T>` is redundant. it would be nice if `getObjectValue` returned T, but doesn't. you can remove the type parameter

pass `expectedValues` as an immutable object (not a Builder, which could be an out-parameter due to it's typical usage)
`ImmutableList.Builder<T> expectedValues` -> `List<?> expectedValues`

Reply:
Thanks for the feedback, adopted the signature